dns/octodns-docker/check-zone.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+'''
+Octo-DNS Reporter / exit non-zero on failure
+'''
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+from concurrent.futures import ThreadPoolExecutor
+from dns.exception import Timeout
+from dns.resolver import NXDOMAIN, NoAnswer, NoNameservers, Resolver, query
+from logging import getLogger
+from sys import stdout,exit
+import re
+
+from octodns.cmds.args import ArgumentParser
+from octodns.manager import Manager
+from octodns.zone import Zone
+
+
+class AsyncResolver(Resolver):
+
+    def __init__(self, num_workers, *args, **kwargs):
+        super(AsyncResolver, self).__init__(*args, **kwargs)
+        self.executor = ThreadPoolExecutor(max_workers=num_workers)
+
+    def query(self, *args, **kwargs):
+        return self.executor.submit(super(AsyncResolver, self).query, *args,
+                                    **kwargs)
+
+
+def main():
+    parser = ArgumentParser(description=__doc__.split('\n')[1])
+
+    parser.add_argument('--config-file', required=True,
+                        help='The Manager configuration file to use')
+    parser.add_argument('--zone', required=True, help='Zone to dump')
+    parser.add_argument('--source', required=True, default=[], action='append',
+                        help='Source(s) to pull data from')
+    parser.add_argument('--num-workers', default=4,
+                        help='Number of background workers')
+    parser.add_argument('--timeout', default=8,
+                        help='Number seconds to wait for an answer')
+    parser.add_argument('server', nargs='+', help='Servers to query')
+
+    args = parser.parse_args()
+
+    manager = Manager(args.config_file)
+
+    log = getLogger('report')
+
+    try:
+        sources = [manager.providers[source] for source in args.source]
+    except KeyError as e:
+        raise Exception('Unknown source: {}'.format(e.args[0]))
+
+    zone = Zone(args.zone, manager.configured_sub_zones(args.zone))
+    for source in sources:
+        source.populate(zone)
+
+    print('name,type,ttl,{},consistent'.format(','.join(args.server)))
+    resolvers = []
+    ip_addr_re = re.compile(r'^[\d\.]+$')
+    for server in args.server:
+        resolver = AsyncResolver(configure=False,
+                                 num_workers=int(args.num_workers))
+        if not ip_addr_re.match(server):
```

Comment:
minor, but I would do this first in the block

Reply:
done